### ISSUE TITLE
Mac Catalyst on macOS does not render well when resizing an app

### ISSUE THREAD

**Issue by ArchieCoder** [unoplatform/uno#16690]
### Current behavior

Resizing the app causes multiple rendering issues

https://github.com/unoplatform/uno/assets/1608424/410f9655-2559-4fea-a1ca-a28c104fc721

### Expected behavior

Rendering should be using full window only

### How to reproduce it (as minimally and precisely as possible)

Contact me for the sample project

### Works on UWP/WinUI

None

### Affected platforms

Mac Catalyst

**Comment by ArchieCoder** [comment]
White using the LiveCharts, I see this bad rendering on MacOS. There is also the black background that should be white like Windows and Linux.

I don't know if it is related with the library, skia or Uno.

![image](https://github.com/unoplatform/uno/assets/1608424/8e3073b4-1827-4e88-91e4-245d10a89588)

**Comment by spouliot** [comment]
Weird, it's like something is not aware of (all?) the resizing.

> I don't know if it is related with the library, skia or Uno.

If it's fine with Windows and Linux then it's probably a Uno, macOS-specific issue.

> Contact me for the sample project

Can you share a test case with me ? Sebastien at platform dot uno

Thanks!

**Comment by agneszitte** [comment]
> Weird, it's like something is not aware of (all?) the resizing.
> 
> > I don't know if it is related with the library, skia or Uno.
> 
> If it's fine with Windows and Linux then it's probably a Uno, macOS-specific issue.
> 
> > Contact me for the sample project
> 
> Can you share a test case with me ? Sebastien at platform dot uno
> 
> Thanks!

@ArchieCoder not sure if you had the chance to send a test case to @spouliot for this issue

**Comment by ArchieCoder** [comment]
@agneszitte the project was sent on May 21st, @spouliot replied: "je regarde ça asap" :)

FYI: Mac Catalyst is not target platform anymore for me when I learned more about it. Skia / Mac does the job.

It's up for the Uno team to close it or not since I won't use catalyst.

**Comment by agneszitte** [comment]
> @agneszitte the project was sent on May 21st, @spouliot replied: "je regarde ça asap" :)
> 
> FYI: Mac Catalyst is not target platform anymore for me when I learned more about it. Skia / Mac does the job.
> 
> It's up for the Uno team to close it or not since I won't use catalyst.

Thank you for the feedback @ArchieCoder, appreciated. 
@spouliot will be able to look at this for Uno depending on the priorities on his side.

**Comment by spouliot** [comment]
I can confirm that the provided code has issues with Catalyst (similar to the original video) but works fine on desktop/macOS.

OTOH the livechart looked good to me.

**Comment by ArchieCoder** [comment]
FYI, I created this GitHub issue just in case it was a bug from the library last month https://github.com/beto-rodriguez/LiveCharts2/issues/1512

**Comment by spouliot** [comment]
Thanks, I'll follow up on that issue for the graph. It seems, at least partially, related to the animation and a later redraw makes it look good (but with a black background).

**Comment by spouliot** [comment]
## Origin Resize Issue

<img width="1919" alt="Screenshot 2024-06-20 at 10 21 47 AM" src="https://github.com/unoplatform/uno/assets/260465/4e727405-5000-4df6-a7b8-0a5981008452">

After some resizing the catalyst app (right) can have a grey bar (incorrect origin?) at the top of the window too.

## Variable Right Grey Band Size

The size of the "grey band" (on the right) is related to showing/hiding the left menu list. Depending on with width of the app's main window, this list is either:
* visible, with both icon and text
* visible, with icon only
* not visible

Changing the state of the list leads to different band size as seen in the next 3 screenshots:

<img width="685" alt="Screenshot 2024-06-20 at 2 57 39 PM" src="https://github.com/unoplatform/uno/assets/260465/6451f0e5-311c-4e6f-a343-ac2a0c7aa571">

<img width="842" alt="Screenshot 2024-06-20 at 2 58 23 PM" src="https://github.com/unoplatform/uno/assets/260465/12570ed9-aa75-4813-b894-2c1e891d9365">

<img width="1428" alt="Screenshot 2024-06-20 at 2 58 59 PM" src="https://github.com/unoplatform/uno/assets/260465/d540af31-c2a3-4010-92c3-b33b119d09ef">


## Fonts issue

The catalyst app does not find the `Segoe UI` fonts (see logs below) while this does not seem to be an issue with the desktop (skia) version.

That can explain why all the text looks smaller on the catalyst version, which in turn makes it show more content (since some elements are smaller). More a _size_ issue than a _resize_ issue.

```
[0:] Windows.UI.UIFontHelper: Warning: Failed to get system font based on Segoe UI-DemiBold
2024-06-20 10:31:38.233 ElementsTechnician[89326:160262623] W/Windows.UI.UIFontHelper: Failed to get system font based on Segoe UI-DemiBold 
[0:] 06-20 10:31:38.233-04:00 W/Windows.UI.UIFontHelper: Failed to get system font based on Segoe UI-DemiBold 
[0:] Windows.UI.UIFontHelper: Warning: Failed to get system font based on XamlAutoFontFamily
```

The lack of this font (on some platforms, like macOS) is a known issue: https://github.com/unoplatform/uno/issues/3954

**Comment by jeromelaban** [comment]
The fonts issue is being address, apps will use OpenSans by default.

**Comment by spouliot** [comment]
The main issue (resize) seems related to the `NavigationView` and how it adapts the sizes of its parts depending on the conditions (width and height).

It has some iOS specific code (that would also be used by Catalyst) but I'm not convinced it's the (only) problem (still next thing to try).

The fact that you don't often resize iOS apps (well never but a bit more often for iPadOS apps) is likely why this was not spotted earlier.

The 2nd issue is very likely unrelated (or I missed something) and might to be a transparency issue - where it should not be transparent and overpaint the whole graph.

**Comment by ajpinedam** [comment]
Hey @ArchieCoder can you please attach/share a sample app or at least the XAML I could use to reproduce this issue? I will be looking into this, following @spouliot findings.

**Comment by ArchieCoder** [comment]
@ajpinedam I sent my project a while ago to @spouliot. If he is not available, send me your email and I will forward the email.

**Comment by spouliot** [comment]
@ArchieCoder just back home, I forwarded your email w/attachment to @ajpinedam

**Comment by ajpinedam** [comment]
Thanks @spouliot  :)

**Comment by beto-rodriguez** [comment]
> updates on live charts do not seem to clear previous drawings (unless a redraw is made)

@spouliot On the LiveCharts side, clearing the canvas is something we do depending on the UI framework, some frameworks require it, some frameworks don't.

In the specific case of Uno, all the supported platforms do not require to clear the canvas, but on MacOS it seems that it behaves different, should this be handled in the LiveCharts side?

**Comment by spouliot** [comment]
> should this be handled in the LiveCharts side?

@beto-rodriguez No. The Mac backend should behave identically with the others. If it does not it's a bug :)

I suspect some _optimizations_ are the problem - it's just not common to see the issue so I have not noticed. I'll look harder :D

**Comment by d2dyno1** [comment]
Any updates on this issue?

**Comment by jeromelaban** [comment]
@d2dyno1 have you been able to test with the `net9.0-desktop` variant instead of using Catalyst? Is there a specific reason you need Catalyst for your app?

**Comment by spouliot** [comment]
@jeromelaban I think @d2dyno1 is referring in a different issue (on `netXX-desktop`) that was mentioned in later, not the original one (on `netXX-maccatalyst`)

I'll double-check the status and, if needed, open a separate issue for it.

**Comment by spouliot** [comment]
new issue @ https://github.com/unoplatform/uno/issues/19675
